extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_2_speech.py

```diff
@@ -0,0 +1,240 @@
+import base64
+import copy
+import io
+import numpy as np
+import torch
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+from transformers import Pipeline, pipeline
+from scipy.io.wavfile import write as write_wav
+
+from aiconfig.default_parsers.parameterized_model_parser import ParameterizedModelParser
+from aiconfig.model_parser import InferenceOptions
+from aiconfig.schema import (
+    ExecuteResult,
+    Output,
+    OutputDataWithValue,
+    Prompt,
+    PromptMetadata,
+)
+from aiconfig.util.params import resolve_prompt
+
+# Circuluar Dependency Type Hints
+if TYPE_CHECKING:
+    from aiconfig.Config import AIConfigRuntime
+
+
+# Step 1: define Helpers
+def refine_pipeline_creation_params(model_settings: Dict[str, Any]) -> List[Dict[str, Any]]:
+    supported_keys = {
+        "torch_dtype",
+        "force_download",
+        "cache_dir",
+        "resume_download",
+        "proxies",
+        "output_loading_info",
+        "local_files_only",
+        "use_auth_token",
+        "revision",
+        "custom_revision",
+        "mirror",
+        "device_map",
+        "max_memory",
+        "offload_folder",
+        "offload_state_dict",
+        "low_cpu_mem_usage",
+        "use_safetensors",
+        "variant",
+    }
+
+    pipeline_creation_params: Dict[str, Any] = {}
+    completion_params: Dict[str, Any] = {}
+    for key in model_settings:
+        if key.lower() in supported_keys:
+            pipeline_creation_params[key.lower()] = model_settings[key]
+        else:
+            if key.lower() == "kwargs" and isinstance(model_settings[key], Dict):
+                completion_params.update(model_settings[key])
+            else:
+                completion_params[key.lower()] = model_settings[key]
+
+    return [pipeline_creation_params, completion_params]
+
+
+def refine_completion_params(unfiltered_completion_params: Dict[str, Any]) -> Dict[str, Any]:
+    supported_keys = {
+        # ???
```

Comment:
You need to go to the public API docs to figure this out. Ex: this is how I got the ones for text2image: https://github.com/lastmile-ai/aiconfig/blob/54de9cc8a1055975b4d28b877bea3dbdc627644e/extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_2_image.py#L82-L84

Reply:
In general for these hugging face model parsers, I start with the general tasks here (https://huggingface.co/tasks) and then look for more details. In this case we want Text-to-Speech one

Reply:
Any idea how to find this for tts?